src/player_display.cpp

```diff
@@ -765,460 +878,560 @@ Strength - 4;    Dexterity - 4;    Intelligence - 4;    Perception - 4" ) );
-                                       get_dodge(), level.level(), level.exercise() < 0 ? 0 : level.exercise() );
-                        } else {
-                            mvwprintz( w_skills, i + 1, 19, cstatus, "%-2d(%2d%%)", level.level(),
-                                       ( level.exercise() <  0 ? 0 : level.exercise() ) );
-                        }
-                    }
-                    wrefresh( w_skills );
-                    line = 0;
-                    curtab++;
-                } else if( action == "CONFIRM" ) {
-                    get_skill_level_object( selectedSkill->ident() ).toggleTraining();
-                } else if( action == "QUIT" ) {
-                    done = true;
-                }
+void player_display_info::draw_inactive_skills(player& player)
+{
+    werase( w_skills );
+    mvwprintz( w_skills, 0, 0, c_light_gray, header_spaces.c_str() );
+    center_print( w_skills, 0, c_light_gray, title_SKILLS );
+    for( size_t i = 0; i < skillslist.size() && i < size_t( skill_win_size_y ); i++ ) {
+        const Skill *thisSkill = skillslist[i];
+        const SkillLevel &level = player.get_skill_level_object( thisSkill->ident() );
+        bool can_train = level.can_train();
+        bool isLearning = level.isTraining();
+        bool rusting = level.isRusting();
+
+        nc_color cstatus;
+        if( rusting ) {
+            cstatus = isLearning ? c_light_red : c_red;
+        } else if( !can_train ) {
+            cstatus = c_white;
+        } else {
+            cstatus = isLearning ? c_light_blue : c_blue;
+        }
+
+        mvwprintz( w_skills, i + 1,  1, cstatus, "%s:", thisSkill->name().c_str() );
+
+        if( thisSkill->ident() == skill_id( "dodge" ) ) {
+            mvwprintz( w_skills, i + 1, 14, cstatus, "%4.1f/%-2d(%2d%%)",
+                       player.get_dodge(), level.level(), level.exercise() < 0 ? 0 : level.exercise() );
+        } else {
+            mvwprintz( w_skills, i + 1, 19, cstatus, "%-2d(%2d%%)", level.level(),
+                       ( level.exercise() <  0 ? 0 : level.exercise() ) );
+        }
+    }
+    wrefresh( w_skills );
+}
+
+void player::disp_info()
+{
+    player_display_info display_info(*this);
+    display_info.draw_borders();
+    input_context ctxt = display_info.draw_header(*this);
+    display_info.draw_inactive_menus(*this);
+
+    // Initial printing is DONE.  Now we give the player a chance to scroll around
+    // and "hover" over different items for more info.
+    do {
+        werase( display_info.w_info );
+        switch( display_info.current_tab ) {
+        case player_display_info::STATS:
+            display_info.draw_active_stats(*this, ctxt);
+            break;
+        case player_display_info::ENCUMBRANCE:
+            display_info.draw_active_encumbrance(*this, ctxt);
+            break;
+        case player_display_info::SKILLS:
+            display_info.draw_active_skills(*this,ctxt);
```

Comment:
Please use proper code style, see `doc/CODE_STYLE.md` or run the code through `astyle`.

Reply:
I ran it through astyle now.